Library/Homebrew/dev-cmd/linkage.rb

```diff
@@ -0,0 +1,127 @@
+#
+# Description: check linkage of installed keg
+# Usage:
+#   brew linkage <formulae>
+#
+# Only works on installed formulae. An error is raised if it is run on uninstalled
+# formulae.
+#
+# Options:
+#  --test      - testing version: only display broken libs; exit non-zero if any
+#                breakage was found.
+
+require "set"
+require "keg"
+require "formula"
+
+module Homebrew
+  def linkage
+    found_broken_dylibs = false
+    ARGV.kegs.each do |keg|
+      ohai "Checking #{keg.name} linkage" if ARGV.kegs.size > 1
+      result = LinkageChecker.new(keg)
+      if ARGV.include?("--test")
+        result.display_test_output
+      else
+        result.display_normal_output
+      end
+      found_broken_dylibs = true unless result.broken_dylibs.empty?
+    end
+    if ARGV.include?("--test") && found_broken_dylibs
+      exit 1
+    end
+  end
+
+  class LinkageChecker
```

Comment:
No problem. But you may want to put this class in a standalone file if it will be used in other part of core in the future.